src_tests/test_network_simulation_strategies.py

```diff
@@ -190,3 +190,42 @@ def test_lossy_network_with_uniform_losses_on_bosonful_modes_and_higher_losses_o
         self._strategy_factory.set_experiment_configuration(self._experiment_configuration)
         self.__check_if_given_distribution_is_close_to_lossy_network_distribution(
             distribution_with_huge_losses_on_bosonless_modes)
+
+    def test_distance_of_generalized_cliffords_with_lossy_network_and_lossy_input(self) -> None:
+        distribution_with_lossy_net = self.__calculate_generalized_cliffords_distribution_with_lossy_network()
+        distribution_with_lossy_input = self.__calculate_generalized_cliffords_distribution_with_lossy_inputs()
+
+        distance_bound_between_estimated_distributions = \
+            self.__calculate_statistical_distance_bound_between_two_approximate_distributions(
+                outcomes_number=self._possible_outcomes_number)
+
+        distance_between_distributions = count_total_variation_distance(
+            distribution_with_lossy_input, distribution_with_lossy_net
+        )
+
+        self.assertLessEqual(distance_between_distributions, distance_bound_between_estimated_distributions)
+
+    def __calculate_generalized_cliffords_distribution_with_lossy_network(self) -> List[float]:
+        """
+                This method calculates approximate distribution for lossy states using generalized cliffords
```

Comment:
This comment seems incorrectly indented.

Reply:
> I wish I worked every day with a code like this...

I appreciate your help. :)